File: hackathon_api/src/main/java/by/mts/hackathon_api/Models/VmModel.java
package by.mts.hackathon_api.Models;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "vm")

public class VmModel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = false, nullable = false)
    private String password;

    @Column(unique = true, nullable = true)
    private Integer port;  

    @Column(unique = true, nullable = false)
    private String name;  

    @Column(unique = false, nullable = true)
    private Long idSsh;
    
    @Column(unique = false, nullable = false)
    private Integer  ram;  

    @Column(unique = false, nullable = false)
    private Integer  rom;  

    @Column(unique = false, nullable = false)
    private Integer cors;

    @Column(unique = false, nullable = false)
    private String  os;

    @Column(unique = false)
    private Integer status;


}

File: hackathon_mts_2026/lib/domain/model/vm_model.dart
class VmModel {
  final int id;
  final String name;
  final String password;
  final int ram;
  final int rom;
  final int cors;
  final String os;
  final int status;

  VmModel({
    required this.id,
    required this.name,
    required this.ram,
    required this.rom,
    required this.cors,
    required this.password,
    required this.status,
    required this.os,
  });

  factory VmModel.fromJson(Map<String, dynamic> json) {
    return VmModel(
      id: json['id'] as int,
      name: json['name'] as String,
      ram: json['ram'] as int,
      rom: json['rom'] as int,
      cors: json['cors'] as int,
      password: json['password'] as String,
      status: json['status'] as int,
      os: json['os'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'ram': ram,
      'rom': rom,
      'cors': cors,
      'password': password,
      'status': status,
      'os': os,
    };
  }
}
